ui/pages/routes/routes.component.tsx

```diff
@@ -624,7 +624,18 @@ export default function Routes() {
             component={ConfirmTransaction}
           />
           <Authenticated path={`${SEND_ROUTE}/:page?`} component={SendPage} />
-          <Authenticated path={SWAPS_ROUTE} component={Swaps} />
+          <Route path={SWAPS_ROUTE}>
+            {(props: RouteComponentProps) => {
+              const { location: v5Location } = props;
+              // eslint-disable-next-line @typescript-eslint/no-explicit-any
+              const SwapsComponent = Swaps as any;
```

Comment:
We could avoid eslint-disable and cast as `any` here with something like this:
```suggestion
              const SwapsComponent = Swaps as React.ComponentType<{
                location: RouteComponentProps['location'];
              }>;
```

The other approach would be having utility type like this:
```typescript
type ComponentWithRouterLocation<TProps = Record<string, unknown>> =
    TProps & {
      location: RouteComponentProps['location'];
    };
```

Then you can reuse it everywhere like this:
```typescript
const SwapsComponent = Swaps as React.ComponentType<ComponentWithRouterLocation>;
```